src/App.tsx

```diff
@@ -1,61 +1,143 @@
+import React, { useState } from 'react';
 import './App.scss';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import { User } from './components/types/User';
+import { Todo } from './components/types/Todo';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+enum FormType {
+  newTitleForm = 'newTitleForm',
+  userSelectForm = 'userSelectForm',
+}
+
+function getUserById(id: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === id);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+type ChangeableElement = HTMLInputElement | HTMLSelectElement;
+
+export const App: React.FC = () => {
+  const [visibleTodos, setTodos] = useState(todos);
+  const [chosenUser, setChosenUser] = useState<User | null>(null);
+  const [newTodoTitle, setTodoTitle] = useState('');
+  const [titleError, setTitleErrorStatus] = useState(false);
+  const [userError, setUserErrorStatus] = useState(false);
+
+  const handleChange = (event: React.ChangeEvent<ChangeableElement>) => {
+    const { name, value } = event.target;
+
+    switch (name) {
+      case FormType.newTitleForm.toString():
+        setTitleErrorStatus(false);
+        setTodoTitle(value.replace(/[^a-zA-Zа-яА-Я0-9\s]/g, ''));
+        break;
+
+      case FormType.userSelectForm.toString():
+        setUserErrorStatus(false);
+        setChosenUser(getUserById(+value));
+        break;
+
+      default: throw new Error('error');
+    }
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!chosenUser) {
+      setUserErrorStatus(true);
+    }
+
+    if (!newTodoTitle.trim()) {
```

Comment:
make variable for it
```suggestion
    if (!...) {
```